packages/devtools-proxy-support/src/fetch.ts

```diff
@@ -37,6 +38,35 @@ async function importNodeFetch(): Promise<typeof fetch> {
 
   return typeof module === 'function' ? module : module.default;
 }
+
+let warningsSuppressed = false;
+
+// In Node.js 23 require()ing ESM modules will work, but emit an experimental warning like
+// CommonJS module ABC is loading ES Module XYZ using require().
+// This function suppresses experimental warnings to avoid those leaking into end users' output.
+function suppressExperimentalWarningsIfNecessary() {
+  const nodeMajorVersion = process.versions.node.split('.').map(Number)[0];
+  if (nodeMajorVersion >= 23 && !warningsSuppressed) {
+    const originalEmit = process.emitWarning;
+    process.emitWarning = (warning, ...args: any[]): void => {
+      if (args[0] === 'ExperimentalWarning') {
+        return;
+      }
+
+      if (
+        typeof args[0] === 'object' &&
+        args[0].type === 'ExperimentalWarning'
+      ) {
+        return;
+      }
+
+      originalEmit(warning, ...args);
+    };
+
+    warningsSuppressed = true;
+  }
+}
```

Comment:
If this is the approach we're going with – which I think is pretty reasonable – then I'd do that in the mongosh CLI itself, like we already disable warnings for the case of the compiled executable in `packages/cli-repl/src/run.ts`, because that's where we're expecting overrides for Node.js behavior and other modifications to truly global state, how does that sound to you?

Reply:
Sure, that sounds reasonable to me! I guess originally I wanted to put it next to the code that does the requiring shenanigans, but you're right that the warnings being displayed or not is better left to the user-facing application.

Reply:
Yeah, I think there's also just going to be _more_ code running into this problem in the future, the more popular ESM gets in our dependencies, not just the code here